conans/client/cmd/uploader.py

```diff
@@ -111,18 +111,32 @@ def _upload(self, conan_file, conan_ref, packages_ids, retry, retry_wait,
                                      "no packages can be uploaded")
             total = len(packages_ids)
             for index, package_id in enumerate(packages_ids):
-                ret_upload_package = self._upload_package(PackageReference(conan_ref, package_id),
+                pref = PackageReference(conan_ref, package_id)
+                cur_package_remote = self._registry.prefs.get(pref)
+
+                if remote_name:  # If remote_name is given, use it
+                    p_remote = self._registry.remotes.get(remote_name)
+                else:
+                    p_remote = cur_package_remote or default_remote
+
+                ret_upload_package = self._upload_package(pref,
                                                           index + 1, total, retry, retry_wait,
                                                           integrity_check,
-                                                          policy, upload_remote)
+                                                          policy, p_remote)
+
+                if (not cur_package_remote or not remote_name) and policy != UPLOAD_POLICY_SKIP:
+                    self._registry.prefs.set(pref, p_remote.name)
```

Comment:
I would want to know why it is conditioned to ``not cur_package_remote or not remote_name``... At first, I expected the registry to be updated unconditionally if the package was uploaded.

Reply:
This is an internal decision (probably arbitrary), that if you are installing from R1 but uploading to R2, you want the updates still from R1, otherwise you need to change the reference manually.